hw/top_earlgrey/data/clocks.xdc

```diff
@@ -21,6 +21,12 @@ create_generated_clock -name clk_io_div2 -source [get_pin ${u_pll}/CLKOUT0] -div
 set u_div4 top_*/u_clkmgr_aon/u_no_scan_io_div4_div
 create_generated_clock -name clk_io_div4 -source [get_pin ${u_pll}/CLKOUT0] -divide_by 4 [get_pin ${u_div4}/u_clk_div_buf/gen_xilinx.u_impl_xilinx/gen_fpga_buf.bufg_i/O]
 
+# the step-down mux is implemented with a LUT right now and the mux switches on the falling edge. 
+# therefore, Vivado propagates both clock edges down the clock network.
+# this implementation is not ideal - but we can at least tell Vivado to only honour the rising edge for 
```

Comment:
> this implementation is not ideal

A curiosity (and not really important): Does our model of a clock divider + mux + buffer in the FPGA include too many ASIC details? Would it have made sense to use a BUFR to divide by 2 and feed that output (+ original) into a BUFGMUX?

Reply:
I believe this has been discussed on this issue https://github.com/lowRISC/opentitan/issues/2676 a while ago, and the decision has been to keep it simple as long as there are no related timing issues (the divided clock is pretty slow).

We may have to revisit this, though, if we see more issues popping up that are related to clock distribution.

Reply:
Fabric-generated clocks are typically not good, clean sources, but as long as it's slow enough to meet timing and we aren't using it for "high-speed" I/O, I guess it should be fine.

Quotes are around it because after working with DRAM interfaces and multi-gigabit transceivers for awhile, I have to remind myself that sometimes 50 MHz is considered high-speed, haha. It all depends on the context.